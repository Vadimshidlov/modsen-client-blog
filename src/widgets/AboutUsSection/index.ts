export { AboutUsSection } from '@/widgets/AboutUsSection/ui/AboutUsSection';
