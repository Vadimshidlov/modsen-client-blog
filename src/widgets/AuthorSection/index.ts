export { AuthorSection } from '@/widgets/AuthorSection/ui/AuthorSection';
