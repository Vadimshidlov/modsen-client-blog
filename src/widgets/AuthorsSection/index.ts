export { AuthorsSection } from '@/widgets/AuthorsSection/ui/AuthorsSection';
