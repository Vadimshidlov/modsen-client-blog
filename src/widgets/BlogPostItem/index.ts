export { BlogPostItem } from '@/widgets/BlogPostItem/BlogPostItem';
