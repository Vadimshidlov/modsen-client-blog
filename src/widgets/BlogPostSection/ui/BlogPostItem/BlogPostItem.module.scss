.blogPostContainer {
  padding-top: 128px;
  display: flex;
  flex-direction: column;
  align-items: center;

  @media (max-width: $tablet) {
    padding-top: 64px;
  }

  .postPreviewBlock {
    display: flex;
    flex-direction: column;
    margin-bottom: 64px;

    @media (max-width: $tablet) {
      align-items: center;
      margin-bottom: 32px;
    }

    @media (max-width: $mobile) {
      margin-bottom: 20px;
    }

    .postPreviewTitle {
      margin-top: 24px;
      margin-bottom: 34px;
      max-width: 768px;
      @include headingOne;

      @media (max-width: $tablet) {
        text-align: center;
        @include headingTwo;
        margin-bottom: 25px;
      }

      @media (max-width: $mobile) {
        @include headingFour;
        margin-bottom: 15px;
      }
    }

    .postCategoryInfo {
      display: flex;
      align-items: center;
      column-gap: 8px;

      .postCategoryImage {
        width: 24px;
        height: 24px;
      }

      .postCategoryName {
        @include headingFour;
      }
    }
  }

  .postImage {
    width: 100%;
    height: 582px;
    object-fit: cover;
    margin-bottom: 64px;

    @media (max-width: $tablet) {
      margin-bottom: 32px;
      height: 400px;
    }

    @media (max-width: $mobile) {
      margin-bottom: 20px;
      height: 350px;
    }
  }

  .postText {
    display: flex;
    flex-direction: column;

    .firstPartTextWrapper {
      display: flex;
      flex-direction: column;
      max-width: 843px;
      margin-bottom: 24px;
    }

    .secondPartTextWrapper {
      display: flex;
      flex-direction: column;
      max-width: 843px;
      margin-bottom: 24px;
    }

    .threePartTextWrapper {
      display: flex;
      flex-direction: column;
      max-width: 843px;
      margin-bottom: 128px;

      @media (max-width: 1000px) {
        margin-bottom: 64px;
      }

      @media (max-width: $tablet) {
        margin-bottom: 32px;
      }

      @media (max-width: $mobile) {
        margin-bottom: 16px;
      }
    }

    .postTextTitle {
      margin-bottom: 16px;
      @include headingTwo;
      letter-spacing: -1px;

      @media (max-width: $tablet) {
        @include headingThree;
      }

      @media (max-width: $mobile) {
        @include headingFour;
      }
    }

    .postTextMessage {
      margin-bottom: 24px;
      max-width: 843px;
      @include bodyOne;
    }

    .postTextList {
      margin-bottom: 24px;

      .postTextListItem {
        @include headingFour;
        list-style: disc inside;

        @media (max-width: $tablet) {
          @include headingFour;
        }

        @media (max-width: $mobile) {
          @include headingSix;
        }
      }
    }
  }
}
