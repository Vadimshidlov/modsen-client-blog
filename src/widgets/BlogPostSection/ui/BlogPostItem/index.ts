export { BlogPostItem } from '@/widgets/BlogPostSection/ui/BlogPostItem/BlogPostItem';
