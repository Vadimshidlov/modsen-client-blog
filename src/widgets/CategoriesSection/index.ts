export { CategoriesSection } from '@/widgets/CategoriesSection/ui/CategoriesSection';
