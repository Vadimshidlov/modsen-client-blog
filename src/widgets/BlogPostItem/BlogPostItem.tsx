import React from 'react';
import Image from 'next/image';
import { POSTS } from '@/shared/constants/posts';
import { AuthorTitleCard } from '@/shared/ui/AuthorTitleCard';
import { AUTHORS_LIST } from '@/shared/constants/authors';
import styles from '@/widgets/BlogPostItem/BlogPostItem.module.scss';

export function BlogPostItem({ postId }: { postId: number }) {
  const selectedPostData = POSTS.filter((postData) => postData.id === postId)[0];
  const { postDate, author, categoryIcon, title, category, img: postImage } = selectedPostData;
  const postAuthorData = AUTHORS_LIST.filter((authorData) => authorData.name === author)[0];
  const { img: authorAvatar, id: authorId } = postAuthorData;

  return (
    <section className={styles.blogPostContainer}>
      <div className={styles.postPreviewBlock}>
        <AuthorTitleCard
          authorId={authorId}
          src={authorAvatar}
          title={author}
          subtitle={`Posted on ${postDate}`}
        />
        <h1 className={styles.postPreviewTitle}>{title}</h1>

        <div className={styles.postCategoryInfo}>
          <Image className={styles.postCategoryImage} src={categoryIcon} alt="category-pic" />
          <span className={styles.postCategoryName}>{category}</span>
        </div>
      </div>
      <Image className={styles.postImage} src={postImage} alt="post-image" />
      <article className={styles.postText}>
        <div className={styles.firstPartTextWrapper}>
          <h2 className={styles.postTextTitle}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.
          </h2>
          <span className={styles.postTextMessage}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
            incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque
            viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus. Sociis natoque
            penatibus et magnis dis parturient montes. Ridiculus mus mauris vitae ultricies leo.
            Neque egestas congue quisque egestas diam. Risus in hendrerit gravida rutrum quisque
            non.
          </span>
        </div>
        <div className={styles.secondPartTextWrapper}>
          <h2 className={styles.postTextTitle}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.
          </h2>
          <span className={styles.postTextMessage}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
            incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque
            viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus. Sociis natoque
            penatibus et magnis dis parturient montes. Ridiculus mus mauris vitae ultricies leo.
            Neque egestas congue quisque egestas diam. Risus in hendrerit gravida rutrum quisque
            non.
          </span>
          <span className={styles.postTextMessage}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
            incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque
            viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus. Sociis natoque
            penatibus et magnis dis parturient montes. Ridiculus mus mauris vitae ultricies leo.
            Neque egestas congue quisque egestas diam. Risus in hendrerit gravida rutrum quisque
            non.
          </span>
          <ul className={styles.postTextList}>
            <li className={styles.postTextListItem}>Lorem ipsum dolor sit amet</li>
            <li className={styles.postTextListItem}>Non blandit massa enim nec scelerisque</li>
            <li className={styles.postTextListItem}>Neque egestas congue quisque egestas</li>
          </ul>
          <span className={styles.postTextMessage}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
            incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque
            viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus. Sociis natoque
            penatibus et magnis dis parturient montes. Ridiculus mus mauris vitae ultricies leo.
            Neque egestas congue quisque egestas diam. Risus in hendrerit gravida rutrum quisque
            non.
          </span>
        </div>
        <div className={styles.threePartTextWrapper}>
          <h2 className={styles.postTextTitle}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.
          </h2>
          <span className={styles.postTextMessage}>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
            incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque
            viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus. Sociis natoque
            penatibus et magnis dis parturient montes. Ridiculus mus mauris vitae ultricies leo.
            Neque egestas congue quisque egestas diam. Risus in hendrerit gravida rutrum quisque
            non.
          </span>
        </div>
      </article>
    </section>
  );
}
