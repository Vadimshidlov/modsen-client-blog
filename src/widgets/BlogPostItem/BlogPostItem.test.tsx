import React from 'react';
import { render } from '@testing-library/react';
import { BlogPostItem } from '@/widgets/BlogPostItem';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('BlogPostItem tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<BlogPostItem postId={1} />);
    expect(container).toMatchSnapshot();
  });
});
