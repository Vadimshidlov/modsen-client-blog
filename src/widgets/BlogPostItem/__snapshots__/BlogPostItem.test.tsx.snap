// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostItem tests should matches snapshot 1`] = `
<div>
  <section
    class="blogPostContainer"
  >
    <div
      class="postPreviewBlock"
    >
      <div
        class="authorInfoBlock"
      >
        <img
          alt="author-avatar"
          class="authorInfoAvatar"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <div
          class="authorInfoText"
        >
          <a
            href="/authors/1"
          >
            <span
              class="authorInfoName"
            >
              Floyd Miles
            </span>
          </a>
          <span
            class="authorInfoAddress"
          >
            Posted on May 23, 2024
          </span>
        </div>
      </div>
      <h1
        class="postPreviewTitle"
      >
        Step-by-step guide to choosing great font pairs
      </h1>
      <div
        class="postCategoryInfo"
      >
        <img
          alt="category-pic"
          class="postCategoryImage"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <span
          class="postCategoryName"
        >
          Technology
        </span>
      </div>
    </div>
    <img
      alt="post-image"
      class="postImage"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <article
      class="postText"
    >
      <div
        class="firstPartTextWrapper"
      >
        <h2
          class="postTextTitle"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.
        </h2>
        <span
          class="postTextMessage"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus. Sociis natoque penatibus et magnis dis parturient montes. Ridiculus mus mauris vitae ultricies leo. Neque egestas congue quisque egestas diam. Risus in hendrerit gravida rutrum quisque non.
        </span>
      </div>
      <div
        class="secondPartTextWrapper"
      >
        <h2
          class="postTextTitle"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.
        </h2>
        <span
          class="postTextMessage"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus. Sociis natoque penatibus et magnis dis parturient montes. Ridiculus mus mauris vitae ultricies leo. Neque egestas congue quisque egestas diam. Risus in hendrerit gravida rutrum quisque non.
        </span>
        <span
          class="postTextMessage"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus. Sociis natoque penatibus et magnis dis parturient montes. Ridiculus mus mauris vitae ultricies leo. Neque egestas congue quisque egestas diam. Risus in hendrerit gravida rutrum quisque non.
        </span>
        <ul
          class="postTextList"
        >
          <li
            class="postTextListItem"
          >
            Lorem ipsum dolor sit amet
          </li>
          <li
            class="postTextListItem"
          >
            Non blandit massa enim nec scelerisque
          </li>
          <li
            class="postTextListItem"
          >
            Neque egestas congue quisque egestas
          </li>
        </ul>
        <span
          class="postTextMessage"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus. Sociis natoque penatibus et magnis dis parturient montes. Ridiculus mus mauris vitae ultricies leo. Neque egestas congue quisque egestas diam. Risus in hendrerit gravida rutrum quisque non.
        </span>
      </div>
      <div
        class="threePartTextWrapper"
      >
        <h2
          class="postTextTitle"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod.
        </h2>
        <span
          class="postTextMessage"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Non blandit massa enim nec. Scelerisque viverra mauris in aliquam sem. At risus viverra adipiscing at in tellus. Sociis natoque penatibus et magnis dis parturient montes. Ridiculus mus mauris vitae ultricies leo. Neque egestas congue quisque egestas diam. Risus in hendrerit gravida rutrum quisque non.
        </span>
      </div>
    </article>
  </section>
</div>
`;
